package_manager/contracts/PR/RepoPackage.sol

```diff
@@ -0,0 +1,98 @@
+pragma solidity ^0.4.24;
+
+
+/**
+ * @title RepoPackage
+ */
+contract RepoPackage {
+  event NewVersion(uint256 versionId, uint16[3] semanticVersion);
+
+  struct Version {
+    uint16[3] semanticVersion;
+    address contractAddress;
+    bytes contentURI;
+  }
+
+  Version[] public versions;
+  mapping (bytes32 => uint256) internal versionIdForSemantic;
+  mapping (address => uint256) internal latestVersionIdForContract;
+
+  function newVersion(uint16[3] _newSemanticVersion, address _contractAddress, bytes _contentURI) public {
+    address contractAddress = _contractAddress;
+    if (versions.length > 0) {
+      Version storage lastVersion = versions[versions.length - 1];
+      require(isValidBump(lastVersion.semanticVersion, _newSemanticVersion));
+      if (contractAddress == 0) {
+        contractAddress = lastVersion.contractAddress;
+      }
+      // Only allows smart contract change on major version bumps
+      require(lastVersion.contractAddress == contractAddress || _newSemanticVersion[0] > lastVersion.semanticVersion[0]);
+    } else {
+      versions.length += 1;
+      uint16[3] memory zeroVersion;
+      require(isValidBump(zeroVersion, _newSemanticVersion));
+    }
+
+    uint256 versionId = versions.push(Version(_newSemanticVersion, contractAddress, _contentURI)) - 1;
```

Comment:
I'm not sure how serious this is, but we're relying on transaction ordering to identify versions, and this might be unreliable because of reorgs.

Can we assess the severity of this and consider alternatives?

One rather efficient alternative would be the concatenation of the version components. This should be unique anyway.

Reply:
Good point, I hadn't thought of that. I was actually thinking about removing versionId altogether, given that there is already the `semanticVersionHash` that could be used, which is similar to the concatenation you mention.

Reply:
Similar but only probabilistically unique!